lib/sentry/opentelemetry/span_storage.ex

```diff
@@ -0,0 +1,151 @@
+defmodule Sentry.OpenTelemetry.SpanStorage do
+  @moduledoc false
+  use GenServer
+
+  defstruct [:cleanup_interval, :table_name]
+
+  alias Sentry.OpenTelemetry.SpanRecord
+
+  @cleanup_interval :timer.minutes(5)
+  @span_ttl :timer.minutes(30)
```

Comment:
Can we refactor these and the options below to use NimbleOptions like the rest of the library? The schema would look something like:

```elixir
@opts [
  cleanup_interval: [
    type: :pos_integer,
    default: :timer.minutes(5)
  ],
  span_ttl: [
    type: :timeout,
    default: :timer.minutes(30)
  ]
]
```